### PR TITLE
Docker update base images

### PR DESCRIPTION
**Pull request by Alkarex**
Hello,
I noticed some Docker base images were getting a bit old, so:

* Update to current versions: python:3.11-slim-bookworm, debian:bookworm-slim, alpine:3.18
* Reduce size of Debian image (remove apt cache) by 10MB
* ~~Add pipefail (which I needed during test), cf. https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/~~
* And also minor Markdown lint of INSTALL.md

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -238,14 +240,15 @@ RUN apk add --no-cache nss
...
+FROM debian:bookworm-slim
+RUN apt-get update && apt-get install -y ca-certificates \
 {{#firefox}}
 # curl tries to load the CA certificates for libnss.
 # It loads them from /usr/lib/libnssckbi.so and /usr/lib/libnsspem.so,
 # which are supplied by 'libnss3' and 'nss-plugin-pem' on debian.
-RUN apt-get install -y libnss3 nss-plugin-pem
+    libnss3 nss-plugin-pem \
 {{/firefox}}
+    && rm -rf /var/lib/apt/lists/*
```
Comment (Alkarex) [inline Dockerfile.template:251]: Delete apt cache (about 10MB)

File: INSTALL.md [hunk 2]
```diff
@@ -3,9 +3,10 @@
 This guide shows how to compile and install curl-impersonate and libcurl-impersonate from source.
 The build process takes care of downloading dependencies, patching them, compiling them and finally compiling curl itself with the needed patches.
 There are currently three build options depending on your use case:
-* [Native build](#Native-build) using an autotools-based Makefile
-* [Cross compiling](#Cross-compiling) using an autotools-based Makefile
-* [Docker container build](#Docker-build)
+
+* [Native build](#native-build) using an autotools-based Makefile
+* [Cross compiling](#cross-compiling) using an autotools-based Makefile
+* [Docker container build](#docker-build)
```
Comment (Alkarex) [inline INSTALL.md:9]: The anchors were slightly wrong (markdown lower-cases them)

File: INSTALL.md [hunk 3]
```diff
@@ -14,7 +15,8 @@ There are two versions of `curl-impersonate` for technical reasons. The **chrome
 ### Ubuntu
 
 Install dependencies for building all the components:
-```
+
+```sh
```
Comment (Alkarex) [inline INSTALL.md:19]: Indication of the language for e.g. syntax highlighting

**Comment by lwthiker** [comment]
Hi,

The build fails due to the Docker change. Is there any point in including it in the final images, if it's used mainly for debugging?

**Comment by Alkarex** [comment]
> The build fails due to the Docker change. Is there any point in including it in the final images, if it's used mainly for debugging?

@lwthiker I have removed the `pipefail` instructions. If you like, I can make a new PR to propose them again. I believe it is useful in the final images to avoid the risks of having some instructions failing and going unnoticed.

**Comment by Alkarex** [comment]
@lwthiker Sorry I had not tested all build combinations. Should be fixed now 🤞🏻

**Comment by wrobelda** [comment]
Not a project owner, but this LGTM.

**Pull request merged**